### PR TITLE
Mod2 complete homework

### PR DESCRIPTION
**Pull request by mariuszlisowski**

### CODE REVIEW

File: module2/homework/generateSequence/generateSequence.hpp [hunk 1]
```diff
@@ -2,6 +2,11 @@
 #include <vector>
 
 std::vector<int> generateSequence(int count, int step) {
-    // TODO: Implement me :)
-    return {};
+	std::vector<int> vec;
+	if (count > 0) {
+		int value {};
+		while (count--)
+			vec.push_back(value += step);
```
Comment (piezol) [inline module2/homework/generateSequence/generateSequence.hpp:9]: Vec might get its memory allocated several times.
We know the size of vector upfront, so we should allocate memory for all elements with reserve function or with constructor.

File: module2/homework/nwdNww/nwdNww.hpp [hunk 2]
```diff
@@ -1,11 +1,18 @@
...
+	}
+	return abs(lhs);
 }
 
 int NWW(int lhs, int rhs) {
-    // TODO: Implement me :)
-    return -1;
+    if (!lhs || !rhs)
+		return 0;
+	return abs(lhs * rhs / NWD(lhs, rhs));
```
Comment (piezol) [inline module2/homework/nwdNww/nwdNww.hpp:17]: I'm copying my comment from different PR:
There are trivial cases that your code won't work for, ie NWW(1e8, 1e8).
That is due to lhs*rhs overflowing int.
You can fix it by casting lhs to unsigned long long in the multiplication.

There are some nontrivial cases that will still fail, ie NWW(1e8-1, 1e8+1), but they fail for the std:lcm(int, int) as well, so it's ok.

Comment (piezol) [reply module2/homework/nwdNww/nwdNww.hpp:17]: Namespace is not specified for abs (you should use std::abs)

File: module2/homework/generateSequence/generateSequence.hpp (hunk 1 above)
Comment (mariuszlisowski) [reply module2/homework/generateSequence/generateSequence.hpp:9]: Yes, indeed. It couldn't be done with the constructor as 'count' can be negative so I used 'reserve' function as suggested (after making sure 'count' was positive). I also tried to use the access operator [] but it doesn't increase the size of a vector (although worked well with the 'resize' function). Thank you piezol!

File: module2/homework/nwdNww/nwdNww.hpp (hunk 2 above)
Comment (mariuszlisowski) [reply module2/homework/nwdNww/nwdNww.hpp:17]: I overlooked that. Thanks for review piezol! Casting used so it shouldn't overflow now. Also appropriate 'absolute' function used for 'long long' (std::llabs). Tested for (1e8, 1e8) but still fails for (1e8-1, 1e8+1) as the function returns int.

```cpp
#pragma once
#include <cstdlib>

int NWD(int lhs, int rhs) {
    int remainder;
    while (rhs !=0) {
        remainder = lhs % rhs;
        lhs = rhs;
        rhs = remainder;
    }
    return std::abs(lhs);
}

int NWW(int lhs, int rhs) {
    if (!lhs || !rhs)
        return 0;
    return std::llabs(static_cast<unsigned long long>(lhs)
                      * rhs / NWD(lhs, rhs));
}
```

File: module2/homework/generateSequence/generateSequence.hpp (hunk 1 above)
Comment (mariuszlisowski) [reply module2/homework/generateSequence/generateSequence.hpp:9]:
```cpp
#pragma once
#include <vector>

std::vector<int> generateSequence(int count, int step) {
    std::vector<int> vec;
    if (count > 0) {
        vec.reserve(count);
        int value {};
        while (count--)
            vec.push_back(value += step);
    }
    return vec;
}
```

**Pull request closed without merging**